AppCenterAuth/AppCenterAuth/MSAuth.m

```diff
@@ -181,14 +182,18 @@ - (void)signInInWithCompletionHandler:(MSSignInCompletionHandler _Nullable)compl
   }
   if ([[MS_Reachability reachabilityForInternetConnection] currentReachabilityStatus] == NotReachable) {
     [self callCompletionHandler:completionHandler
-                  withErrorCode:MSACAuthErrorSignInWhenNoConnection
+                  withErrorCode:MSACAuthErrorNoConnection
                      andMessage:@"User sign-in failed. Internet connection is down."];
     return;
   }
   if (self.clientApplication == nil || self.authConfig == nil) {
-    [self callCompletionHandler:completionHandler
-                  withErrorCode:MSACAuthErrorSignInBackgroundOrNotConfigured
-                     andMessage:@"signIn is called while it's not configured or not in the foreground."];
+    if (self.signInShouldWaitForConfig) {
+      MSLogDebug([MSAppCenter logTag], @"Downloading configuration in process. Waiting for it before sign-in.");
+    } else {
+      [self callCompletionHandler:completionHandler
+                    withErrorCode:MSACAuthErrorSignInBackgroundOrNotConfigured
+                      andMessage:@"signIn is called while it's not configured or not in the foreground."];
+    }
     return;
```

Comment:
Shouldn't this `return` kept inside of `else` block? Note that we're returning directly without  signing in when `self.signInShouldWaitForConfig` is true, which it's not the right behavior.

Reply:
Yep, that's OK. It will continue after downloading the config